### PR TITLE
79 add loading error overlay

### PR DESCRIPTION
**Pull request by dennis95stumm**
# Fixes \#79

## Type (Highlight the corresponding type)
- Bugfix
- Feature
- Documentation
- **Refactoring (e.g. Style updates, Test implementation, etc.)**
- Other (please describe):

## Checklist
- [x] Code updated to current masters head
- [x] Passes CI checks
- [x] Is a part of an issue
- [x] Tests added for the bugfix or newly implemented feature, describe below why if not
- [ ] Changelog is updated
- [x] Documentation of code and features exists

## Changes
- Added loading overlay for roles view page
- Added error handling during load and persist for roles view page
- Added loading overlay for users view page
- Added error handling during load and persist for users view page
- Added check that only the profile page for the own user can be opened

### CODE REVIEW

File: resources/js/views/settings/users/View.vue [hunk 1]
```diff
@@ -332,12 +361,25 @@ export default {
   mounted () {
     EventBus.$on('currentUserChangedEvent', this.togglePermissionFlags);
 
-    if (this.config.type !== 'profile') {
+    if (!['profile', 'view'].includes(this.config.type)) {
```
Comment (SamuelWei) [inline resources/js/views/settings/users/View.vue:364]: you also don't need to load the roles if the user is editing himself (not via the profile but via the user page)

**Review by SamuelWei — CHANGES_REQUESTED** [review]

File: resources/js/views/settings/users/View.vue [hunk 2]
```diff
@@ -447,13 +489,18 @@ export default {
           this.$store.dispatch('session/logout').then(() => {
             this.$router.push({ name: 'home' });
           });
+          Base.error(error, this.$root, error.message);
         } else if (error.response && error.response.status === env.HTTP_UNPROCESSABLE_ENTITY) {
           this.errors = error.response.data.errors;
```
Comment (SamuelWei) [inline resources/js/views/settings/users/View.vue:494]: if there is an error with the roles field, you could trigger a reload of the roles. I did something similar on my open PR

Comment (dennis95stumm) [reply resources/js/views/settings/users/View.vue:494]: So should I change here something?

File: resources/js/views/settings/users/View.vue [hunk 3]
```diff
@@ -361,7 +361,14 @@ export default {
   mounted () {
     EventBus.$on('currentUserChangedEvent', this.togglePermissionFlags);
 
-    if (!['profile', 'view'].includes(this.config.type)) {
+    if (
```
Comment (SamuelWei) [inline resources/js/views/settings/users/View.vue:364]: the rule is not working correctly. I found many cases where it was loading the data even if not necessary. E.g. why would you want to load the permissions if the user can't edit them?

**Review by SamuelWei — CHANGES_REQUESTED** [review]

File: resources/js/views/settings/users/View.vue (hunk 3 above)
Comment (SamuelWei) [reply resources/js/views/settings/users/View.vue:364]: I think this should work
```js
 if (this.config.id === 'new' || (
      PermissionService.can('editUserRole', {
        id: this.config.id,
        model_name: 'User'
      }) && this.config.type === "edit")
    ) {
      this.loadRoles();
    }
```

Comment (dennis95stumm) [reply resources/js/views/settings/users/View.vue:364]: Yeah I've used the wrong method. This should be can!

**Review by SamuelWei — APPROVED** [review]

**Pull request merged**